.github/workflows/build_test_linux_fedora.yaml

```diff
@@ -27,15 +33,13 @@ jobs:
     - name: Build dpsimpy and test examples
       shell: bash
       working-directory: ${{ github.workspace }}/build
-      run: cmake --build . --target dpsimpy --target tests --target dpsimpyvillas
-      env:
-        MAKEFLAGS: "-j2"
+      run: cmake --build . --target dpsimpy --target tests --target dpsimpyvillas --parallel $(nproc)
```

Comment:
I'm not sure if this is still the case but it used to be that `nproc` within a container would report the cores of the host rather than the cores assigned to the container. The runner is executed in a container, so `nproc` might not report the correct number.
Maybe we should parse `/sys/fs/cgroup/cpuset/cpuset.cpus` instead? Or set a fixed number of threads.

Now that I think about it, which runner do we actually use? The default github actions runner? Then this might not be an issue.

Reply:
Hi, we are using the github actions runner... For this case seems to give the correct data

Reply:
Another option would be to use this action [here](https://github.com/SimenB/github-actions-cpu-cores)

```yaml
    steps:
      - name: Get number of CPU cores
        uses: SimenB/github-actions-cpu-cores@v2
        id: cpu-cores
      - name: run tests
        run: npx jest --max-workers ${{ steps.cpu-cores.outputs.count }}
```